### DISCUSSION TITLE
migrating Spacy 2 packages

### DISCUSSION THREAD

**Discussion by DSLituiev** [explosion/spaCy#7403] [Help: Coding & Implementations] [answered]
Is  there  a way to use / convert earlier packages? 

I am struggling with salvaging a package created with Spacy 2.3.1:
1. if I try to load it in Spacy 3, it complains there is no `config.cfg` -- not backward compatible
2. if I install it with  `pip install .`, it's not forward compatible, and reinstalls Spacy 2.3.1

Please help

**Comment by ines** [comment, marked as answer]
Packages trained with spaCy v2.x are not compatible with spaCy v3.x, so you'll have to retrain your model using the new version. There have been many changes to the internals and components to support the new features like transformer-based pipelines, custom models, and so on, so the weights can't be cross-compatible. Also see #7054 for more background on this.

To retrain your model, you can use `spacy init config` to auto-generate a config file for the components you're using, `spacy convert` to convert your data and `spacy train` to train your pipeline.
